examples/lock-app/nrfconnect/main/AppTask.cpp

```diff
@@ -533,27 +536,38 @@ void AppTask::DispatchEvent(AppEvent * aEvent)
 
 void AppTask::UpdateClusterState(BoltLockManager::State state, BoltLockManager::OperationSource source)
 {
-    DlLockState lockState;
+    DlLockState newLockState;
+    chip::app::DataModel::Nullable<chip::app::Clusters::DoorLock::DlLockState> currentLockState;
 
     switch (state)
     {
     case BoltLockManager::State::kLockingCompleted:
-        lockState = DlLockState::kLocked;
+        newLockState = DlLockState::kLocked;
         break;
     case BoltLockManager::State::kUnlockingCompleted:
-        lockState = DlLockState::kUnlocked;
+        newLockState = DlLockState::kUnlocked;
         break;
     default:
-        lockState = DlLockState::kNotFullyLocked;
+        newLockState = DlLockState::kNotFullyLocked;
         break;
     }
 
-    SystemLayer().ScheduleLambda([lockState, source] {
-        LOG_INF("Updating LockState attribute");
+    chip::app::Clusters::DoorLock::Attributes::LockState::Get(kLockEndpointId, currentLockState);
 
-        if (!DoorLockServer::Instance().SetLockState(kLockEndpointId, lockState, source))
-        {
-            LOG_ERR("Failed to update LockState attribute");
-        }
-    });
+    if (currentLockState.IsNull())
+    {
+        // Initialize lock state with start value, but not invoke lock/unlock.
+        chip::app::Clusters::DoorLock::Attributes::LockState::Set(kLockEndpointId, newLockState);
```

Comment:
Access of the attribute storage should also happen on the CHIP thread (within lambda passed to ScheduleLambda).